certificate-renewal/REPWRITERSPECS/BANNO.CDRENEW.V1.POW

```diff
@@ -1732,4 +1911,107 @@ PROCEDURE VALIDATESYSMEMOMODE
   END
 END [PROCEDURE]
 
+PROCEDURE BNOPRINTPROGRAMINFO
+[* Print program info to direct JSON output
+**
+**  PASS    BNOPROGRAMNAME     Program name
+**  PASS    BNOPROGRAMVERSION  Program version
+**  PASS    BNOLASTMODDATE     Last mod date
+**  PASS    BNOLASTMODTIME     Last mod time
+*]
+ PRINT BNOINDENT(1)+","
+ NEWLINE
+ PRINT BNOINDENT(1)+Q+"programInfo"+Q+":"+Q+
+                    BNOPROGRAMNAME+"  "+
+                    BNOPROGRAMVERSION+"  "+
+                    FORMAT("99/99/99 ",BNOLASTMODDATE)+
+                    BNOLASTMODTIME+Q
+END [PROCEDURE]
+
+PROCEDURE BNOPRINTRGDATA
+[* Print @RGUSER[CHR,NUM] data passed from the UX in proper JSON format
+**
+**  PASS    BNODEBUGMODE     Boolean: Print debug data, T/F
+*]
+
+ IF BNODEBUGMODE=BNODEBUGMODEON THEN
+  DO
+   PRINT ","
+   NEWLINE
+   PRINT BNOINDENT(1)+Q+"rgData"+Q+": ["
+   NEWLINE
+   PRINT BNOINDENT(2)+Q+"CHR1:'"+@RGUSERCHR1+"' "+
+                     "CHR2:'"+@RGUSERCHR2+"'"+Q+","
+   NEWLINE
+   PRINT BNOINDENT(2)+Q+"CHR3:'"+@RGUSERCHR3+"' "+
+                     "CHR4:'"+@RGUSERCHR4+"'"+Q+","
+
+   NEWLINE
+   PRINT BNOINDENT(2)+Q+"CHR5:'"+@RGUSERCHR5+"' "+
+                     "NUM1:"+FORMAT("#########9 ",@RGUSERNUM1)+
+                     "NUM2:"+FORMAT("#########9 ",@RGUSERNUM2)+
+                     "NUM3:"+FORMAT("#########9 ",@RGUSERNUM3)+
+                     "NUM4:"+FORMAT("#########9 ",@RGUSERNUM4)+
+                     "NUM5:"+FORMAT("#########9",@RGUSERNUM5)+Q
+   NEWLINE
+   PRINT BNOINDENT(1)+"]"
+   NEWLINE
+  END
+END [PROCEDURE]
+
+PROCEDURE BNODEBUGMSGADD
+[* Add a message line to the debug message array
+**
+** PASS    BNODEBUGMSGMAX     Current array index max lines allowed
+** PASS    BNODEBUGMSG        Debug message to add to array
+**
+** RETURN  BNODEBUGCOUNT      Count of debug message lines
+** RETURN  BNODEBUGMSGLIST(A) Debug lines of data to print
+*]
+ IF BNODEBUGCOUNT<BNODEBUGMSGMAX THEN
+  DO
+   BNODEBUGCOUNT=BNODEBUGCOUNT+1
+   BNODEBUGMSGLIST(BNODEBUGCOUNT)=BNODEBUGMSG
+   BNODEBUGMSG=""
+  END
+END [PROCEDURE]
+
+PROCEDURE BNOPRINTDEBUG
+[* Print debug output in proper JSON format. Requires BNODEBUGMODE to be
+** on (or true).
+**
+**  PASS  BNODEBUGMODE        Boolean: Print debug data, T/F
+**  PASS  BNODEBUGMSGLIST(A)  Array of debug messages to print
+**  PASS  BNODEBUGCOUNT       Number of debug lines to print
+*]
+ IF BNODEBUGMODE=BNODEBUGMODEON THEN
+  DO
+   PRINT ","
+   NEWLINE
+   PRINT BNOINDENT(2)+Q+"debugData"+Q+": ["
+   NEWLINE
+   PRINT BNOINDENT(2)+Q+"Debug note: Debug lines truncated to 126 characters"+Q
+
+   FOR BNODEBUGLOOP=1 TO BNODEBUGCOUNT
+    DO
+     IF BNODEBUGLOOP=1 THEN
```

Comment:
Can you look at the situation in BNOPRINTDEBUG when Debug lines only equal 1 in Line 1995? It seems the code does not put a comma and does not put a new Line if BNODEBUGCOUNT is only 1.  It makes sense if there is enough room to put in 126 characters after the comment before is Printed saying "Debug note: Debug lines truncated to 126 characters".
So that is a bit confusing to me if not.

I noticed in Skip a Pay for the same procedure it automatically puts in  a comma and a new line even if BNODEBUGOUNT is 1.  

Here is the code you have for this in Skip a Pay.
PROCEDURE BNOPRINTDEBUG
[* Print debug output in proper JSON format. Requires DEBUGMODE to be
** on (or true).
**
**  PASS  DEBUGMODE           Boolean: Print debug data, T/F
**  PASS  BNODEBUGMSGLIST(A)  Array of debug messages to print
**  PASS  BNODEBUGCOUNT       Number of debug lines to print
*]

 IF BNODEBUGMODE=BNODEBUGMODEON THEN
  DO
   PRINT ","
   NEWLINE
   PRINT BNOINDENT(2)+Q+"debugData"+Q+": ["
   NEWLINE
   PRINT BNOINDENT(2)+Q+"Debug note: Debug lines truncated to 120 characters"+Q
   IF BNODEBUGCOUNT>0 THEN
    DO
     FOR BNODEBUGLOOP=1 TO BNODEBUGCOUNT
      DO
       PRINT ","
       NEWLINE
       PRINT BNOINDENT(3)+Q+[FORMAT("999-",BNODEBUGLOOP)+]Segment(BNODEBUGMSGLIST(BNODEBUGLOOP),1,120)+Q
      END [FOR BNODEBUGLOOP=1 TO BNODEBUGCOUNT]
    END [IF BNODEBUGCOUNT>0 THEN]
   PRINT BNOINDENT(2)+"]"
   NEWLINE
  END
END [PROCEDURE]

Reply:
Corrected